### ISSUE TITLE
Question about LDM support

### ISSUE THREAD

**Issue by GiannisPikoulis** [ChenWu98/cycle-diffusion#13]
Hello and thanks for sharing your code.

I want to use your implementation with LDM models. I have already trained two LDMs on two separate (but related) domains. How should I proceed so as to perform Image translation between these two domains using my pretrained LDM models.

Thanks in advance.

**Comment by ChenWu98** [comment]
Hi, 

Sorry about the late reply. I have just uploaded the support for LDM. More specifically, it is a demo for translating from FFHQ to CelebAHQ, using the DDIM sampler with eta = 0.1. After pulling the latest code, you can run it using the following command. 
```
export CUDA_VISIBLE_DEVICES=0
export RUN_NAME=translate_ffhq256_to_celeba256_latentdiff_ddim_eta01
export SEED=42
nohup python -m torch.distributed.launch --nproc_per_node 1 --master_port 1498 main.py --seed $SEED --cfg experiments/$RUN_NAME.cfg --run_name $RUN_NAME$SEED --logging_strategy steps --logging_first_step true --logging_steps 4 --evaluation_strategy steps --eval_steps 50 --metric_for_best_model CLIPEnergy --greater_is_better false --save_strategy steps --save_steps 50 --save_total_limit 1 --load_best_model_at_end --gradient_accumulation_steps 4 --num_train_epochs 0 --adafactor false --learning_rate 1e-3 --do_eval --output_dir output/$RUN_NAME$SEED --overwrite_output_dir --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --eval_accumulation_steps 4 --ddp_find_unused_parameters true --verbose true > $RUN_NAME$SEED.log 2>&1 &
```
The output will be saved under `output/translate_ffhq256_to_celeba256_latentdiff_ddim_eta01`, and it should be exactly the same with the following image: 
![eval_000000](https://user-images.githubusercontent.com/28187501/211129743-d383ab1c-67b8-4222-8147-425cac14e4e8.png)

If your custom models are trained on other domains, you may need to modify (1) the test data loader and (2) the code for loading model weights. 

Please let me know if you encounter any problem!

**Comment by ChenWu98** [comment]
Also, do not forget to download the model weights: 
```
cd ckpts/
wget https://www.dropbox.com/s/9lpdgs83l7tjk6c/ldm_models.zip
unzip ldm_models.zip
```

**Comment by GiannisPikoulis** [comment]
Thank you very much for answering my question. 

Is it also possible to perform class-to-class image editing within the same dataset? Lets say you have one dataset with two classes of animals, cats and dogs and the diffusion model (a single LDM) is trained using some class-label conditioning. Is it possible to perform image-editing using a single LDM model in order to edit, let's say a photo of a cat to look like a dog? Would I run forward DPM-Encoder using the 'cat-conditioning' and decode it using the 'dog-conditioning'?

Thanks in advance and I appreciate the time you spend for answering these.

**Comment by ChenWu98** [comment]
Yes this is definitely possible! You can view such class-conditioning as a trivial case of text-conditioning. I didn’t implement this variant in the code, but you may slightly modify the code and use LDMs trained on, say, class-conditioned ImageNet.
